collect_app/src/main/java/org/odk/collect/android/forms/Form.java

```diff
@@ -136,7 +137,7 @@ public Builder jrVersion(String jrVersion) {
         }
 
         public Builder formFilePath(String formFilePath) {
-            this.formFilePath = new StoragePathProvider().getFormDbPath(formFilePath);
```

Comment:
It felt wrong that the `Form` object should have to care about storage details, so I removed this dependency. This also made it simpler to write tests for the `InMemFormsRepository` (which was my clue that something could be improved).